File: src/main/java/com/footverse/cart/dto/CartItemResponse.java
package com.footverse.cart.dto;

import java.math.BigDecimal;

/**
 * One line of the caller's cart (dto-spec §12). Every product field is resolved from the
 * {@code ProductVariantService} purchase snapshot at read time — nothing here is stored on the
 * cart row itself, so a renamed product or a changed price is reflected immediately.
 *
 * @param id               the cart item id
 * @param productVariantId the variant id this line holds
 * @param productId        the owning product id
 * @param productName      the owning product's name
 * @param productImageUrl  the owning product's primary image URL, or {@code null} when it has none
 * @param size             the variant size
 * @param unitPrice        the effective unit price, resolved by the product module
 * @param quantity         the quantity in the cart
 * @param lineTotal        {@code unitPrice × quantity}, computed by the server
 * @param available        {@code false} when the variant is inactive or out of stock
 */
public record CartItemResponse(
        Long id,
        Long productVariantId,
        Long productId,
        String productName,
        String productImageUrl,
        String size,
        BigDecimal unitPrice,
        int quantity,
        BigDecimal lineTotal,
        boolean available) {
}

File: app/lib/features/cart/models/cart_item_response.dart
import 'package:json_annotation/json_annotation.dart';

import '../../product/models/decimal_json.dart';

part 'cart_item_response.g.dart';

/// One line in the caller's cart (dto-spec §12). The money fields
/// ([unitPrice], [lineTotal]) are server-computed and rendered as delivered;
/// the client never recomputes them (dto-spec §1; business-rules). [available]
/// is false when the variant is inactive or out of stock.
@JsonSerializable(createToJson: false)
class CartItemResponse {
  const CartItemResponse({
    required this.id,
    required this.productVariantId,
    required this.productId,
    required this.productName,
    required this.productImageUrl,
    required this.color,
    required this.size,
    required this.unitPrice,
    required this.quantity,
    required this.lineTotal,
    required this.available,
  });

  factory CartItemResponse.fromJson(Map<String, dynamic> json) =>
      _$CartItemResponseFromJson(json);

  final int id;
  final int productVariantId;
  final int productId;
  final String productName;
  final String? productImageUrl;
  final String color;
  final String size;
  @JsonKey(fromJson: decimalFromJson)
  final double unitPrice;
  final int quantity;
  @JsonKey(fromJson: decimalFromJson)
  final double lineTotal;
  final bool available;
}
